libs/block_generator/include/zkevm_framework/block_generator/block_generator.hpp

```diff
@@ -4,14 +4,24 @@
 #include <boost/json/value.hpp>
 #include <expected>
 #include <istream>
+#include <sstream>
 #include <string>
 
 #include "zkevm_framework/data_types/block.hpp"
 
 namespace data_types::block_generator {
 
     // Construct Block from JSON config file
-    std::expected<Block, std::string> generate_block(const boost::json::value &json_value);
+    std::expected<Block, std::string> generate_block(const boost::json::value& json_value);
+
+    class sequence_generator {
+      public:
+        sequence_generator(const std::string& config_file_name);
+        int generate_block(unsigned char* block_data);
```

Comment:
Not sure is `lib/block_generator` is a right place for this function. Filling raw memory with block data is a very specific case, and I hardly could imagine another usage of it. Could you consider moving this code directly into the emulator?

Reply:
ok

Reply:
Well, it this case let's adjust the code to be a part of the library. I think that the interface should return the errors rather then print them to standard output. In some use cases the output could be unavailable. Especially if we consider multi-language execution.

Reply:
result < 0 means error. For using by `cgo` we have to keep interface as much easy as possible. By the way, most probably all block generator will be replaced by pure GO implementation

Reply:
> For using by cgo we have to keep interface as much easy as possible.

Disagree here, it is a responsibility of the wrapper library, not of the block generator. But OK, you can keep error code for now. And I still think that it's better to remove printing errors from block generator. I can't remember any production library that writes anything to `stderr`.